### PR TITLE
Add support for Musl libc

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Since Musl is sufficiently different from Glibc (see https://wiki.musl-libc.org/functional-differences-from-glibc.html), it requires a different import, which now should be applied to files that have `import Glibc` in them.

### CODE REVIEW

**Review by neonichu — APPROVED** [review]
I kinda hate that we have to do this kind of dance for very basic operations like exiting a process (!) or signal handling. This feels like something swift-system should provide abstractions for tbh.

**Comment by MaxDesiatov** [comment]
> This feels like something swift-system should provide abstractions for tbh.

100%, but until then we have to import platform-specific libcs explicitly.

File: Sources/Basics/Cancellator.swift [hunk 1]
```diff
@@ -77,27 +77,22 @@ public final class Cancellator: Cancellable {
...
                 #else
-                var action = sigaction()
                 action.__sigaction_handler = unsafeBitCast(
                     SIG_DFL,
                     to: sigaction.__Unnamed_union___sigaction_handler.self
                 )
+                #endif
                 sigaction(SIGINT, &action, nil)
                 kill(getpid(), SIGINT)
-                #endif
```
Comment (tomerd) [inline Sources/Basics/Cancellator.swift:100]: this diff looks like it is doing more than just Musl support?

Comment (al45tair) [reply Sources/Basics/Cancellator.swift:100]: I think this is just code rearrangement; every case did

```
sigaction(SIGINT, &action, nil)
kill(getpid(), SIGINT)
```

and I think Max just moved it outside of the `#if` conditions.

Comment (MaxDesiatov) [reply Sources/Basics/Cancellator.swift:100]: Yes, those lines are common for all cases, they only differ in handler setup.

Comment (tomerd) [reply Sources/Basics/Cancellator.swift:100]: ah I see the `#endif` moved 👍

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by finagolfin** [comment]
> 100%, but until then we have to import platform-specific libcs explicitly.

[This has already been pitched](https://forums.swift.org/t/pitch-the-cstdlib-module/51373). Would you take that over, @MaxDesiatov, since you are adding Musl, including adding Bionic and the BSD libcs?

I'll be happy to help split off Bionic, I just don't want to push all this through the Evolution process.

**Pull request merged**